IOTConnectSDK/iotconnect-sdk_2_1/src/main/java/com/iotconnectsdk/utils/SDKClientUtils.kt

```diff
@@ -179,4 +221,60 @@ internal object SDKClientUtils {
         }
         return true
     }
+
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    @Throws(Exception::class)
+    fun generateSasTokenLatest(resourceUri: String?, key: String?): String? {
+        // Token will expire in one year
+        val expiry = Instant.now().epochSecond + 31536000
+        val stringToSign = URLEncoder.encode(
+            resourceUri,
+            "UTF-8"
+        ) + "\n" + expiry
+        val decodedKey = java.util.Base64.getDecoder().decode(key)
+        val sha256HMAC = Mac.getInstance("HmacSHA256")
+        val secretKey =
+            SecretKeySpec(decodedKey, "HmacSHA256")
+        sha256HMAC.init(secretKey)
+        val encoder = java.util.Base64.getEncoder()
+        val signature = String(
+            encoder.encode(
+                sha256HMAC.doFinal(stringToSign.toByteArray(StandardCharsets.UTF_8))
+            ), StandardCharsets.UTF_8
+        )
+        return ("SharedAccessSignature sr=" + URLEncoder.encode(resourceUri, "UTF-8")
+                + "&sig=" + URLEncoder.encode(
+            signature,
+            StandardCharsets.UTF_8.name()
+        ) + "&se=" + expiry)
+    }
+
+    @Throws(java.lang.Exception::class)
+    fun generateSasToken(resourceUri: String?, key: String?): String? {
+        // Token will expire in one year
+        val expiry = Instant.now().epochSecond + 31536000
```

Comment:
@sw-tt-rajkiranvank @sw-tt-tusharojha same here

Reply:
@sw-ot-deveshmevada
Issue resolved
Value taken in constant